aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelTree.java

```diff
@@ -95,10 +96,18 @@
         for (Map.Entry<ModelSchema, Set<String>> entry : modelMap.entrySet()) {
             ModelSchema schema = entry.getKey();
             for (String id : entry.getValue()) {
-                // Create dummy model instance using just the ID and model type
-                String dummyJson = gson.toJson(Collections.singletonMap("id", id));
-                Model dummyItem = gson.fromJson(dummyJson, schema.getModelClass());
-                descendants.add(dummyItem);
+                if (root.iterator().next().getClass() == SerializedModel.class) {
```

Comment:
Calling `next()` on the `iterator()` is going to increment the iterator, which I don't think you want to do here, right?  

To fix this, when we call `root.iterator.next()` on line 88 above, we need to save the result to a instance variable, and then use it here.

Reply:
You are right, will update and push a new PR. Thanks for catching this.

Reply:
You may not even need this `if` statement at all. The returned item from this method is used as a method of publishing to remote only, so you can always return a `List<SerializedModel>`.